lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -94,7 +94,10 @@ def uri2indirection(http_method, uri, params)
     end
 
     configured_environment = Puppet.lookup(:environments).get(environment)
-    unless configured_environment.nil?
+    if configured_environment.nil?
+      Puppet.warning("Environment #{environment} does not exist, falling back to #{Puppet[:default_agent_environment]}")
+      params[:environment] = configured_environment = Puppet.lookup(:environments).get(Puppet[:default_agent_environment])
```

Comment:
If the `default_agent_environment` doesn't exist, should we raise? Or let the request pass through with an environment we know doesn't exist? Today we do the latter...

Reply:
We should probably let the request pass through for now to avoid having too many changes, we can always update that afterwards if need be